virtool/data/events.py

```diff
@@ -203,6 +203,13 @@ async def __anext__(self) -> Event:
             try:
                 received = await self._channel.get_json()
 
+                if received is None:
+                    logger.error(
+                        "Channel has been unsubscribed, trying to re-subscribe"
+                    )
+                    (self._channel,) = await self._redis.subscribe(self._channel_name)
+                    continue
+
```

Comment:
Remove this. We are already handling `ChannelClosedError` more rigourously below.

This problem here is that `None` is being put in the queue. Update the `emit` function such that it avoids ever putting `None` in the queue. Log a warning if it is attempted so we can fix the problem.